src/ocean/core/array/Search.d

```diff
@@ -1013,6 +1013,101 @@ unittest
     test!("==")( countIf( "gbbbi"[], ( char c ) { return c == 'd'; } ), 0 );
 }
 
+/*******************************************************************************
+
+    Checks if array B is a subset of array A. Array A needs to be sorted.
+    Array B doesn't need to be sorted and might have duplicate elements. It
+    checks if array A contains each item of array B by using bcontains function.
+    If T is a class or struct, comparison is performed using T.opCmp().
+    Otherwise, elements of T are compared using ">" and ">=" or, if T is
+    compatible to size_t (which includes ssize_t, the signed version of size_t),
+    by calculating the difference.
+
+    Template params:
+        T = type of array element
+
+    Params:
+        a = array A
+        b = array B
+
+    Returns:
+        false if there is any item of array B, which is not an item of array A
+
+    In:
+        a.length must be at most ssize_t.max (int.max if size_t is uint or
+        long.max if size_t is ulong). TODO: Remove this restriction by
+        rephrasing the implementation in bsearchCustom().
+
+*******************************************************************************/
+
+public bool bsubset ( T ) ( T[] a, T[] b )
+{
+    if ( a.length == 0 )
+        return b.length == 0;
+
+    foreach ( ref T item; b )
+    {
+        if ( !bcontains(a, item) )
```

Comment:
Note that this is `O(N * log N)`. It can be optimized to `O(N)` if both A and B are required to be sorted.

Not sure if this is of any importance for use case you have in mind.

Reply:
AFAIU `mismatch` which is implemented in this module is doing that already if the arrays are both sorted and don't have duplicate items. I needed a function which handles the case when I don't have the array B as sorted.

Reply:
Aha, maybe cross-reference these 2 functions in their documentation?